scripts/vad_overlap_posterior.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import glob
+import os
+import time
+from argparse import ArgumentParser
+from itertools import repeat
+from multiprocessing import Pool
+
+import numpy as np
+import pandas as pd
+
+
+"""
+This script serves two purposes:
+ 
+    1) gen_overlap_seq: 
+        Generate predictions with overlapping input segments by using the frame level prediction from NeMo/examples/asr/vad_infer.py. 
+        Then a smoothing filter is applied to decide the label for a frame spanned by multiple segments. 
+       
+    2）gen_seg_table: 
+        Converting frame level prediction to speech/no-speech segment in start and end times format.
+   
+Usage:
+
+python vad_overlap_posterior.py --gen_overlap_seq --gen_seg_table --frame_folder=<FULL PATH OF YOU STORED FRAME LEVEL PREDICTION> --method='median' --overlap=0.875 --num_workers=20 --threshold=0.8 
+ 
+"""
+
+
+def gen_overlap_seq(frame_filepath, per_args):
+    """
+    Given a frame level prediction, generate predictions with overlapping input segments by using it
+    Args:
+        frame_filepath : frame prediction file to be processed.
+        per_args:
+            method : Median or mean smoothing filter.
+            overlap : Amounts of overlap.
+            seg_len : Length of window for generating the frame.
+            shift_len : Amount of shift of window for generating the frame.
+            out_dir : Output dir of generated prediction.
+    """
+
+    try:
+        method = per_args['method']
+        overlap = per_args['overlap']
+        seg_len = per_args['seg_len']
+        shift_len = per_args['shift_len']
+        out_dir = per_args['out_dir']
+
+        frame = np.loadtxt(frame_filepath)
+        name = os.path.basename(frame_filepath).split(".frame")[0] + "." + method
+        overlap_filepath = os.path.join(out_dir, name)
+
+        shift = int(shift_len / 0.01)  # number of units of shift
+        seg = int((seg_len / 0.01 + 1))  # number of units of each window/segment
+
+        jump_on_target = int(seg * (1 - overlap))  # jump on target generated sequence
+        jump_on_frame = int(jump_on_target / shift)  # jump on input frame sequence
+
+        if jump_on_frame <= 1:
+            raise ValueError("Not valid jump on frame sequence. Try different shift and overlap choices")
```

Comment:
how do users know what jump means? may be different error message would be helpful

Reply:
okay good point. will give a more detailed error message. well I didn't expect and want user to worry about how to perform overlap prediction but yes it should be clearer.